sdk/servicebus/azure-messaging-servicebus/README.md

```diff
@@ -100,92 +130,163 @@ ServiceBusReceiverAsyncClient receiver = new ServiceBusClientBuilder()
+[`ServiceBusReceiverClient`][ServiceBusReceiverClient] to receive messages. Each receiver can consume messages from
+either a queue or a topic subscription.
+
+#### Receive a batch of messages
+
+The snippet below creates a `ServiceBusReceiverClient` to receive messages from a topic subscription. It returns a batch
+of messages when 10 messages are received or until 30 seconds have elapsed, whichever happens first.
 
-<!-- embedme ./src/samples/java/com/azure/messaging/servicebus/ReadmeSamples.java#L35-L41 -->
+<!-- embedme ./src/samples/java/com/azure/messaging/servicebus/ReadmeSamples.java#L81-L94 -->
 ```java
-String connectionString = "<< CONNECTION STRING FOR THE SERVICE BUS NAMESPACE >>";
-ServiceBusReceiverAsyncClient receiver = new ServiceBusClientBuilder()
-    .connectionString(connectionString)
+ServiceBusReceiverClient receiver = new ServiceBusClientBuilder()
+    .connectionString("<< CONNECTION STRING FOR THE SERVICE BUS NAMESPACE >>")
     .receiver()
     .topicName("<< TOPIC NAME >>")
     .subscriptionName("<< SUBSCRIPTION NAME >>")
+    .receiveMode(ReceiveMode.PEEK_LOCK)
+    .buildClient();
+
+IterableStream<ServiceBusReceivedMessageContext> messages = receiver.receive(10, Duration.ofSeconds(30));
+messages.forEach(context -> {
+    ServiceBusReceivedMessage message = context.getMessage();
+    System.out.printf("Id: %s. Contents: %s%n", message.getMessageId(),
+        new String(message.getBody(), StandardCharsets.UTF_8));
+});
+```
+
+#### Receive a stream of messages
+
+The asynchronous `ServiceBusReceiverAsyncClient` continuously fetches messages until the `subscription` is disposed.
+
+<!-- embedme ./src/samples/java/com/azure/messaging/servicebus/ReadmeSamples.java#L101-L115 -->
+```java
+ServiceBusReceiverAsyncClient receiver = new ServiceBusClientBuilder()
+    .connectionString("<< CONNECTION STRING FOR THE SERVICE BUS NAMESPACE >>")
+    .receiver()
+    .queueName("<< QUEUE NAME >>")
     .buildAsyncClient();
+
+Disposable subscription = receiver.receive().subscribe(context -> {
+    ServiceBusReceivedMessage message = context.getMessage();
+    System.out.printf("Id: %s%n", message.getMessageId());
+    System.out.printf("Contents: %s%n", new String(message.getBody(), StandardCharsets.UTF_8));
+}, error -> {
+        System.err.println("Error occurred while receiving messages: " + error);
+    }, () -> {
+        System.out.println("Finished receiving messages.");
+    });
+```
+
+### Settle messages
+
+When a message is received, it can be settled using any of the `complete()`, `abandon()`, `defer()`, or `deadLetter()`
+overloads. The sample below completes a received message from synchronous `ServiceBusReceiverClient`.
+
+<!-- embedme ./src/samples/java/com/azure/messaging/servicebus/ReadmeSamples.java#L130-L135 -->
+```java
+receiver.receive(10).forEach(context -> {
+    ServiceBusReceivedMessage message = context.getMessage();
```

Comment:
Is it worth showing how to check the context for an error?

Reply:
It's in a sample. Not part of a mainstream scenario